src/common/config.h

```diff
@@ -90,8 +92,15 @@ struct md_config_t {
   /**
    * The configuration schema, in the form of Option objects describing
    * possible settings.
+   *
+   * flat_map<> is employed to lower memory fragmentation and reduce
+   * CPU cache misses. It still offers logarithmic search complexity
+   * while the extra overhead on insert isn't a problem as modifying
+   * the schema is intended to be rare.
    */
-  std::map<std::string, const Option &> schema;
+  boost::container::flat_map<
```

Comment:
[I can see significant overhead from the `md_config_t::get_val<T>` across the read path](https://gist.github.com/rzarzynski/baba188854f585afad7071c903e93d8d).  The `flat_map` didn't help much. We could try to apply `sstring` but I guess it still will be costly. 

Maybe the solution is to store configurable's value locally (cache it) while having a config observer to keep the copy in pace. We could have this machinery encapsulated as a generic utility to easily apply the pattern in critical places.